File: App/app/src/main/java/com/cinema/ticket_booking/data/repository/NotificationRepository.java
package com.cinema.ticket_booking.data.repository;

import androidx.lifecycle.*;
import com.cinema.ticket_booking.data.model.response.*;
import com.cinema.ticket_booking.network.ApiService;
import com.cinema.ticket_booking.util.Resource;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class NotificationRepository {
    private final ApiService api;

    @Inject
    public NotificationRepository(ApiService api) {
        this.api = api;
    }

    public LiveData<Resource<PageResponse<NotificationResponse>>> getNotifications(int page) {
        MutableLiveData<Resource<PageResponse<NotificationResponse>>> r = new MutableLiveData<>();
        r.setValue(Resource.loading());
        api.getNotifications(page, 20).enqueue(new Callback<>() {
            @Override
            public void onResponse(Call<ApiResponse<PageResponse<NotificationResponse>>> c,
                    Response<ApiResponse<PageResponse<NotificationResponse>>> res) {
                if (res.isSuccessful() && res.body() != null && res.body().isSuccess())
                    r.setValue(Resource.success(res.body().getData()));
                else
                    r.setValue(Resource.error("Tải thông báo thất bại"));
            }

            @Override
            public void onFailure(Call<ApiResponse<PageResponse<NotificationResponse>>> c, Throwable t) {
                r.setValue(Resource.error("Lỗi kết nối: " + t.getMessage()));
            }
        });
        return r;
    }

    public void markAllAsRead() {
        api.markAllAsRead().enqueue(new Callback<>() {
            @Override
            public void onResponse(Call<ApiResponse<Void>> c, Response<ApiResponse<Void>> r) {
            }

            @Override
            public void onFailure(Call<ApiResponse<Void>> c, Throwable t) {
            }
        });
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/repository/NotificationRepository.kt
package com.cinema.ticket_booking.data.repository

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.cinema.ticket_booking.data.model.response.ApiResponse
import com.cinema.ticket_booking.data.model.response.NotificationResponse
import com.cinema.ticket_booking.data.model.response.PageResponse
import com.cinema.ticket_booking.network.ApiService
import com.cinema.ticket_booking.util.Resource
import javax.inject.Inject
import javax.inject.Singleton
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

@Singleton
class NotificationRepository @Inject constructor(
    private val api: ApiService
) {

    fun getNotifications(page: Int): LiveData<Resource<PageResponse<NotificationResponse>>> {
        val r = MutableLiveData<Resource<PageResponse<NotificationResponse>>>()
        r.value = Resource.loading()
        api.getNotifications(page, 20).enqueue(object : Callback<ApiResponse<PageResponse<NotificationResponse>>> {
            override fun onResponse(
                call: Call<ApiResponse<PageResponse<NotificationResponse>>>,
                res: Response<ApiResponse<PageResponse<NotificationResponse>>>
            ) {
                if (res.isSuccessful && res.body() != null && res.body()!!.success) {
                    r.value = Resource.success(res.body()!!.data)
                } else {
                    r.value = Resource.error("Tải thông báo thất bại")
                }
            }

            override fun onFailure(call: Call<ApiResponse<PageResponse<NotificationResponse>>>, t: Throwable) {
                r.value = Resource.error("Lỗi kết nối: ${t.message}")
            }
        })
        return r
    }

    fun markAllAsRead() {
        api.markAllAsRead().enqueue(object : Callback<ApiResponse<Void>> {
            override fun onResponse(c: Call<ApiResponse<Void>>, r: Response<ApiResponse<Void>>) {}
            override fun onFailure(c: Call<ApiResponse<Void>>, t: Throwable) {}
        })
    }

    fun deleteNotification(id: String) {
        api.deleteNotification(id).enqueue(object : Callback<ApiResponse<Void>> {
            override fun onResponse(c: Call<ApiResponse<Void>>, r: Response<ApiResponse<Void>>) {}
            override fun onFailure(c: Call<ApiResponse<Void>>, t: Throwable) {}
        })
    }
}
